lib/add-channel.js

```diff
@@ -1,52 +1,85 @@
-const debug = require('debug')('semantic-release:github');
-const {RELEASE_NAME} = require('./definitions/constants');
-const parseGithubUrl = require('./parse-github-url');
-const resolveConfig = require('./resolve-config');
-const getClient = require('./get-client');
-const isPrerelease = require('./is-prerelease');
-
-module.exports = async (pluginConfig, context) => {
+import debugFactory from "debug";
+
+import { RELEASE_NAME } from "./definitions/constants.js";
+import parseGithubUrl from "./parse-github-url.js";
+import resolveConfig from "./resolve-config.js";
+import isPrerelease from "./is-prerelease.js";
+import { toOctokitOptions, SemanticReleaseOctokit } from "./octokit.js";
+
+const debug = debugFactory("semantic-release:github");
+
+export default async function addChannel(
+  pluginConfig,
+  context,
+  { Octokit = SemanticReleaseOctokit } = {}
```

Comment:
i like the dependency injection approach, but i'm wondering about providing defaults in cases like this. are there any places where this is called without providing an instance?
```suggestion
  { Octokit }
```

Reply:
Good point! Initially I didn't pass an Octokit constructor from `/index.js`, but now that is the case. I'll make the change